python/lsst/ip/isr/calibType.py

```diff
@@ -205,6 +218,63 @@ def updateMetadata(self, camera=None, detector=None, filterName=None,
         mdSupplemental.update(kwargs)
         mdOriginal.update(mdSupplemental)
 
+    def calibInfoFromDict(self, dictionary):
+        """Handle common keywords.
+
+        This isn't an ideal solution, but until all calibrations
+        expect to find everything in the metadata, they still need to
+        search through dictionaries.
+
+        Parameters
+        ----------
+        dictionary : `dict` or `lsst.daf.base.PropertyList`
+            Source for the common keywords.
+
+        Raises
+        ------
+        RuntimeError :
+            Raised if the dictionary does not match the expected OBSTYPE.
+
+        """
+
+        def search(haystack, needles):
+            """Search dictionary 'haystack' for an entry in 'needles'
+            """
+            test = [haystack.get(x) for x in needles]
+            test = set([x for x in test if x is not None])
+            if len(test) == 0:
+                if 'metadata' in haystack:
+                    return search(haystack['metadata'], needles)
+                else:
+                    return None
+            elif len(test) == 1:
+                value = list(test)[0]
+                if value == '':
+                    return None
+                else:
+                    return value
+            else:
+                raise ValueError(f"Too many values found: {len(test)} {test} {needles}")
+
+        if 'metadata' in dictionary:
+            metadata = dictionary['metadata']
+
+            if self._OBSTYPE != metadata['OBSTYPE']:
+                raise RuntimeError(f"Incorrect calibration supplied.  Expected {self._OBSTYPE}, "
+                                   f"found {metadata['OBSTYPE']}")
+
+        self._instrument = search(dictionary, ['INSTRUME', 'instrument'])
+        self._raftName = search(dictionary, ['RAFTNAME'])
+        self._slotName = search(dictionary, ['SLOTNAME'])
+        self._detectorId = search(dictionary, ['DETECTOR', 'detectorId'])
+        self._detectorName = search(dictionary, ['DET_NAME', 'DETECTOR_NAME', 'detectorName'])
+        self._detectorSerial = search(dictionary, ['DET_SER', 'DETECTOR_SERIAL', 'detectorSerial'])
+        self._filter = search(dictionary, ['FILTER', 'filterName'])
```

Comment:
Is this the `raw_md` here, _i.e._ the untranslated header value? Or will this have been through `astro_metadata_translator` and obs_package translation at this point?

Reply:
If the latter then that's fine, if the former then this will cause problems.

Reply:
This is an untranslated header value.  However, it's the untranslated header value of the calibration product, which is derived either from a user specified value (`updateMetadata(filterName='myFavoriteFilter')`), or is set by pulling the value from the input exposure, which should have the translated version.  My goal of including this was to allow validation (something we don't do yet) to confirm that the calibration matches the input data.